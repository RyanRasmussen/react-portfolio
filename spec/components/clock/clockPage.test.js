import React from 'react';
import Clock from '../../../src/components/clock/clockPage';
import { shallow } from 'enzyme';

test('Clock component renders the correct date', () => {
    const date1 = new Date('July 8, 2019 21:18:00');

    const component = shallow(
        <Clock time={date1}></Clock>
    );
    expect(component).toMatchSnapshot();

});