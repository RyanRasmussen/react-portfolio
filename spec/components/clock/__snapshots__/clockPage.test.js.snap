// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clock component renders the correct date 1`] = `
<Container
  tag="div"
>
  <Layout />
  <div
    className="lcars-column full-centered"
  >
    <div
      className="title"
    >
      <h1>
        Hello world, the current time is:
      </h1>
    </div>
    <div
      className="clock-container"
    >
      <div
        className="clock-face lcars-row centered"
      >
        9:18:00 PM
      </div>
      <div
        className="lcars-row centered"
      >
        <div
          className="timer-btn-container lcars-row"
        >
          <div
            className="lcars-column full-centered fill"
          >
            <div
              className="lcars-row centered"
            >
              <div
                className="lcars-row left-btn right"
              >
                <ActionButton
                  color="golden-tanoi"
                  label="Start"
                  onClick={[Function]}
                  otherClass="timer-btn"
                  shape="rounded"
                />
              </div>
              <div
                className="lcars-row right-btn"
              />
            </div>
            <div
              className="clock-face lcars-column centered"
            >
              <div
                className="timer-face lcars-u-4"
              >
                0:0.000
              </div>
            </div>
            <div
              className="lcars-column centered"
            >
              <div
                className="lcars-row"
              >
                <t
                  allowSameDay={false}
                  className="lcars-element lcars-dodger-blue-bg right-rounded date-picker"
                  dateFormat="MM/dd/yyyy"
                  dateFormatCalendar="LLLL yyyy"
                  disabled={false}
                  disabledKeyboardNavigation={false}
                  dropdownMode="scroll"
                  inlineFocusSelectedMonth={false}
                  monthsShown={1}
                  nextMonthButtonLabel="Next Month"
                  nextYearButtonLabel="Next Year"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClickOutside={[Function]}
                  onFocus={[Function]}
                  onInputClick={[Function]}
                  onInputError={[Function]}
                  onKeyDown={[Function]}
                  onMonthChange={[Function]}
                  onSelect={[Function]}
                  onYearChange={[Function]}
                  preventOpenOnFocus={false}
                  previousMonthButtonLabel="Previous Month"
                  previousYearButtonLabel="Previous Year"
                  readOnly={false}
                  renderDayContents={[Function]}
                  selected={2019-07-09T03:18:00.000Z}
                  shouldCloseOnSelect={true}
                  showMonthYearPicker={false}
                  showQuarterYearPicker={false}
                  showTimeInput={false}
                  showTimeSelect={false}
                  strictParsing={false}
                  timeCaption="Time"
                  timeInputLabel="Time"
                  timeIntervals={30}
                  withPortal={false}
                />
              </div>
              <div
                className="lcars-row"
              >
                Today's date - 
                July
                 
                8
                , 
                2019
                 - is 
                0
                 day
                s
                 away from the selected date.
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</Container>
`;
